### ISSUE TITLE
kNN works differently in scikit-learn 1.0

### ISSUE THREAD

**Issue by markotoplak** [biolab/orange3#5610]
scikit-learn introduces some changes. @janezd fixed the tests to pass in #5608.

The failed ROC test points out some changes in the underlying learners. Here I tried replicating the old version of the test in workflow with old scikit. This one seems to correspond to what the tests were expecting.

![image](https://user-images.githubusercontent.com/552182/134958416-ef84e788-7ad1-4a29-8ab2-b7dea88d6991.png)

And here is how it looks with new scikit-learn.

![image](https://user-images.githubusercontent.com/552182/134958575-2dab085a-60f8-4587-a7d1-02567fa4bbb9.png)

I find differences in kNN disturbing. It would be prudent to also check other learners.

**Comment by markotoplak** [comment]
Hm, I saw the following warning in the scikit docs: 

**Regarding the Nearest Neighbors algorithms, if two neighbors and have identical distances but different labels, the result will depend on the ordering of the training data.**

I tried playing with shuffling the data in the following workflow and saw that both the old and new versions are sensitive to the order of the data. The degenerate case we can observe on 1.0 on titanic disappears when the data is shuffled. Makes sense: titanic contains many repeated rows and is nicely sorted.

![image](https://user-images.githubusercontent.com/552182/134961447-b29141c1-5f5e-4aec-8304-4e3a7d0db28c.png)

So perhaps we don't need to worry about it.
